### ISSUE TITLE
Enable non-root sessions

### ISSUE THREAD

**Issue by bryteise** [cloud-hypervisor/libvirt#22]
Currently libvirt is running cloud-hypervisor as root and fails to launch otherwise. Need to move to non-root usage of cloud-hypervisor instead.

**Comment by sboeuf** [comment]
What's the reason CH is failing to run without root permissions? The ones that comes to mind are either `/dev/kvm` is not in the right group or the libvirt user is not part of this group, preventing it from accessing the KVM device. Or maybe you're missing some capabilities for creating the TAP device (`CAP_NET_ADMIN` should work).

**Comment by bryteise** [comment]
This is because libvirt only will run cloud-hypervisor if it is root (hard check in the current implementation). Resolving that part is simple but I need to ensure it works with the new daemon implementation.

**Comment by sboeuf** [comment]
@bryteise oh I see, are you saying that the current libvirt/ch driver is obsolete? And that we have to rely on a "new" way of writing libvirt drivers, which translates into "we need a dedicated daemon"?

**Comment by bryteise** [comment]
Yes, basically #24 needs to be implemented so the ch driver works like the qemu driver.

**Comment by sboeuf** [comment]
Gotcha!
